### PR TITLE
feat(dam): Save columns and aggregates to widget

### PR DESCRIPTION
**Pull request by shruthilayaj**
Adds columns and aggregates to DashboardWidgetQuery
writes. These fields will eventually be written separately
with the new widget building experience and so we will
not need to use the `getColumnsAndAggregates` hack.

Note: still need to add some integration tests to this PR.

Follow up: Update widget library widgets and template
widgets to have columns and aggregates separately.

### CODE REVIEW

**Review by matejminar — APPROVED** [review]
Looks good to me.

In metrics naming convention `field` will go into `aggregates` and `groupBy` will go into `columns`.

File: tests/js/spec/components/modals/addDashboardWidgetModal.spec.jsx [hunk 1]
```diff
@@ -308,6 +308,25 @@ describe('Modals -> AddDashboardWidgetModal', function () {
     wrapper.unmount();
   });
 
+  it('updates widget aggregate', async function () {
```
Comment (priscilawebdev) [inline tests/js/spec/components/modals/addDashboardWidgetModal.spec.jsx:311]: do you think we have similar tests in the widget builder test file too?

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 2]
```diff
@@ -115,23 +116,29 @@ type State = {
   selectedDashboard?: SelectValue<string>;
 };
 
-const newDiscoverQuery = {
+const newDiscoverQuery: WidgetQuery = {
   name: '',
   fields: ['count()'],
+  columns: [],
+  aggregates: ['count()'],
```
Comment (priscilawebdev) [inline static/app/components/modals/addDashboardWidgetModal.tsx:123]: should we add these changes to the widgetBuilder too?

Comment (shruthilayaj) [reply static/app/components/modals/addDashboardWidgetModal.tsx:123]: I'm going to do that today!

**Pull request merged**